src/app/users/login.ts

```diff
@@ -57,18 +58,8 @@ export const login = async ({
 
   if (user instanceof CouldNotFindUserFromPhoneError) {
     subLogger.info({ phone }, "new user signup")
-
     const userRaw: NewUserInfo = { phone: phoneNumberValid, phoneMetadata: null }
-
-    const carrierInfo = await TwilioClient().getCarrier(phoneNumberValid)
-    if (carrierInfo instanceof Error) {
-      // non fatal error
-      subLogger.warn({ phone }, "impossible to fetch carrier")
-    } else {
-      userRaw.phoneMetadata = carrierInfo
-    }
-
-    user = await userRepo.persistNew(userRaw)
+    user = await createUser(userRaw)
```

Comment:
we need to add account creation in `createUser` or `createAccount` and pass accountId to `createUser`

Reply:
Account and User are still the same entity currently. This PR is about separating User/Account from Wallet.

Reply:
I think I understand what you mean. Splitting this further so that the Wallet is only being created from create-account?